qoqo_for_braket/qoqo_for_braket/queued_results.py

```diff
@@ -155,3 +156,79 @@ def poll_result(
             else:
                 return None
         return self._results
+
+
+class QueuedProgramRun:
+    def __init__(self, measurement, queued_circuits: List[QueuedCircuitRun]) -> None:
+        self._measurement = measurement
+        self._queued_circuits: List[QueuedCircuitRun] = queued_circuits
+        self._results: Optional[Dict[str, float]] = None
+
+    def poll_result(self):
+        registers: List[
+            Dict[str, List[List[bool]]],
+            Dict[str, List[List[float]]],
+            Dict[str, List[List[complex]]],
+        ] = [
+            {},
+            {},
+            {},
+        ]  # just empty registers
+        all_finished = [False] * len(self._queued_circuits)
+        while not all(all_finished):
+            for i, queued_circuit in enumerate(self._queued_circuits):
+                res = queued_circuit.poll_result()
+                if res is not None:
+                    registers[0].update(res[0])  # add results to bit registers
+                    registers[1].update(res[1])  # add results to float registers
+                    registers[2].update(res[2])  # add results to complex registers
+                    all_finished[i] = True
```

Comment:
We should not have a while loop here, but simply return none if the runs have not finished yet. Since running on remote can potentially take long, the user should decide how often to retry.